src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -574,9 +561,11 @@ public List<ClusterTransactionCommand.SingleClusterDatabaseCommand> ExecuteClust
                     }
 
                     ExecuteClusterTransactionOneByOne(batch);
+                    ClusterTransactionIndexWaiter.SetAndNotifyListenersIfHigher(maxIndex);
```

Comment:
why not to place this in `OnClusterTransactionCompletion` ?

Reply:
The max can be of a cluster transaction command with no database command so the `OnClusterTransactionCompletion` will not be called.
I filtered those commands and I retrieved only the max index because that is the relevant part for us to update the last cluster transaction index